### PR TITLE
log/transport: alternative Identity for NetMQTransport log

### LINKED ISSUE #2738: Set identity of DealerSocket as MessageRequest.Id

### PR DESCRIPTION
**Pull request by colibrishin**
Closes #2738.

### CODE REVIEW

**Comment by longfin** [comment]
- My intention was, to leverage this `GUID` for `DealerSocket`'s identifier. (it can be set through `dealerSocket.Options.Identity = req.Id.ToByteArray()`).
- In addition, specifying identifiers in a content-addressable manner, as in the current implementation, can be problematic from a strict point of view, because even if the message has exactly the same content, it is necessary to distinguish each message when it is sent twice or more.

**Review by longfin — DISMISSED** [review]

**Comment by longfin** [comment]
Well, maybe we need a changelog? (or need to skip explicitly via commit message).

**Comment by colibrishin** [comment]
> Well, maybe we need a changelog? (or need to skip explicitly via commit message).

I think this is a minor change to be added. I'll skip the changelog.

**Review by longfin — APPROVED** [review]

File: Libplanet.Net/Transports/NetMQTransport.cs [hunk 1]
```diff
@@ -600,11 +600,13 @@ private void ReceiveMessage(object sender, NetMQSocketEventArgs e)
                             try
                             {
                                 Message message = _messageCodec.Decode(copied, false);
+                                Guid reqId = new Guid(copied[0].ToByteArray());
```
Comment (longfin) [inline Libplanet.Net/Transports/NetMQTransport.cs:603]: Ah... we can't assume the first frame can be translated as GUID always, so need handling for that. 😢

**Review by longfin — CHANGES_REQUESTED** [review]

File: Libplanet.Net/Transports/NetMQTransport.cs (hunk 1 above)
Comment (colibrishin) [reply Libplanet.Net/Transports/NetMQTransport.cs]: Added a fallback exception If `Identity` cannot be parsed as GUID, create a new one.

File: Libplanet.Net/Transports/NetMQTransport.cs [hunk 2]
```diff
@@ -599,14 +599,37 @@ private void ReceiveMessage(object sender, NetMQSocketEventArgs e)
                         {
                             try
                             {
+                                Guid reqId;
```
Comment (longfin) [inline Libplanet.Net/Transports/NetMQTransport.cs:602]: I think that `reqId` will be used only for logging purposes, so turning it to`string` and parse like as below would be enough.

```csharp
string requestId;
try
{
    requestId =  new Guid(copied[0].ToByteArray()).ToString();
}
catch (ArgumentException)
{
    requestId = "unknown";
}
```

**Review by longfin — APPROVED** [review]

**Comment by riemannulus** [comment]
Could you rebase this PR, please?

**Comment by colibrishin** [comment]
Rerouting branch to use next internal testnet.

**Review by greymistcube — APPROVED** [review]

**Pull request merged**